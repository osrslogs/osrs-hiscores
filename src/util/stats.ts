import { RankedStats } from '../types';

/**
 * Finds the name of the first ranked stat in rankedStats
 *
 * @param {RankedStats} rankedStats
 *
 * @returns {string | undefined} The name of the ranked stat, else undefined
 */
// eslint-disable-next-line import/prefer-default-export
export const getRankedStat = (rankedStats: RankedStats): string | undefined => {
  // eslint-disable-next-line no-restricted-syntax
  for (const [key, value] of Object.entries(rankedStats)) {
    if (value.rank > 0) {
      return key;
    }
  }
  return undefined;
};
