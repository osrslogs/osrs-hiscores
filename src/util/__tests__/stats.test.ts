import { getRankedStat } from '../stats';
import statsConfig from '../test-util/stats-mock.json';

describe('getRankedSkill', () => {
  it('returns a skill name when there is a ranked skill', () => {
    const result = getRankedStat(statsConfig.rankedSkill);
    expect(result).toMatchSnapshot();
  });

  it('returns undefined when there are no ranked skills', () => {
    const result = getRankedStat(statsConfig.noRankedSkill);
    expect(result).toBe(undefined);
  });
});
