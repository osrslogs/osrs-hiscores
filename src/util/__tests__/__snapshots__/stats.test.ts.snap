// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getRankedSkill returns a skill name when there is a ranked skill 1`] = `"construction"`;
